kyaml/setters2/settersutil/substitutioncreator.go

```diff
@@ -62,3 +73,106 @@ func (c SubstitutionCreator) Create(openAPIPath, resourcesPath string) error {
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 }
+
+// CreateSettersForSubstitution creates the setters for all the references in the substitution
+// values if they don't already exist in openAPIPath file.
+func (c SubstitutionCreator) CreateSettersForSubstitution(openAPIPath string) error {
+	b, err := ioutil.ReadFile(openAPIPath)
```

Comment:
consider using https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/yaml/types.go#L123

Reply:
I think this method is more explicit is what I feel.

Reply:
At least use this: https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/yaml/types.go#L104

No need to copy-paste these lines around.

Reply:
Ahh thanks for pointing.